### PR TITLE
Improve Haskell highlighter

### PR DESCRIPTION
**Pull request by danr**
Import keywords are put in keyword face instead of meta face.
This leaves room for pragmas and macros to be in the meta face.

Operator keywords are put in keyword face too.

Finally, expression keywords are put in face attribute.

### CODE REVIEW

File: rc/base/haskell.kak [hunk 1]
```diff
@@ -12,19 +12,53 @@ hook global BufCreate .*[.](hs) %{
...
-add-highlighter -group /haskell/code regex \b(as|case|class|data|default|deriving|do|else|hiding|if|in|infix|infixl|infixr|instance|let|module|newtype|of|qualified|then|type|where)\b 0:keyword
-add-highlighter -group /haskell/code regex \b(Int|Integer|Char|Bool|Float|Double|IO|Void|Addr|Array|String)\b 0:type
+add-highlighter -group /haskell/code regex (?<=\W)0x+[A-Fa-f0-9]+ 0:value
+add-highlighter -group /haskell/code regex (?<=\W)\d+([.]\d+)? 0:value
+add-highlighter -group /haskell/code regex \b(import|hiding|qualified)\b 0:keyword
+add-highlighter -group /haskell/code regex \b(import|module)\b[^\n]+\b(as)\b 1:keyword 2:keyword
+add-highlighter -group /haskell/code regex \b(class|data|default|deriving|infix|infixl|infixr|instance|module|newtype|pattern|type|where)\b 0:keyword
+add-highlighter -group /haskell/code regex \b(case|do|else|if|in|let|mdo|of|proc|rec|then)\b 0:attribute
+
+                                 # matches uppercase identifiers:  Monad Control.Monad
```
Comment (lenormf) [inline rc/base/haskell.kak]: All the comments in the hunks are indented by several levels, is it on purpose?

Comment (mawww) [inline rc/base/haskell.kak]: Yeah, this commenting style looks a bit odd

Comment (mawww) [inline rc/base/haskell.kak]: ping

Comment (danr) [inline rc/base/haskell.kak]: Oh, I think it was when `add-highlighter` was `addhl`, they were aligned to the start of the regular expression.

Comment (mawww) [inline rc/base/haskell.kak]: Any chance to have an updated version of this PR with comments properly aligned ?

Comment (danr) [inline rc/base/haskell.kak]: Yes, I'll look into it

**Pull request closed without merging**